### PR TITLE
[DYN-3027] Libraries core node model coverage

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Increase code coverage for the Libraries/CoreNodeModel project

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )
Michael Kirschner (@mjkkirschner )

### FYIs

Alfredo Pozo (@alfredo-pozo )

### CODE REVIEW

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Castle.Core, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc, processorArchitecture=MSIL">
+      <HintPath>..\..\..\src\packages\Castle.Core.3.3.3\lib\net45\Castle.Core.dll</HintPath>
```
Comment (QilongTang) [inline test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:35]: I did not see this dependency before, is this a new one or? Did I miss something?

File: test/Libraries/CoreNodeModelsTests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (QilongTang) [inline test/Libraries/CoreNodeModelsTests/Properties/AssemblyInfo.cs:35]: Need changes here, I believe all binaries in Dynamo project all use same version strategy

Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/Properties/AssemblyInfo.cs]: Thanks, I missed that, its changed already

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj (hunk 1 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj]: You are right, I checked and dont know why that was added. I removed it with no issues

File: test/Libraries/CoreNodeModelsTests/packages.config [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Castle.Core" version="3.3.3" targetFramework="net48" />
```
[the remark this answers is not in the document]
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/packages.config]: ready

File: src/Dynamo.All.sln [hunk 4]
```diff
@@ -235,6 +235,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "IronPythonExtension", "Iron
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "PythonMigrationViewExtension", "PythonMigrationViewExtension\PythonMigrationViewExtension.csproj", "{10AF430D-0D3A-49CE-A63D-848912959745}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CoreNodeModelsTests", "..\test\Libraries\CoreNodeModelsTests\CoreNodeModelsTests.csproj", "{F9C5DFBD-CB56-4331-B2AD-6390A8D29CA0}"
```
Comment (mjkkirschner) [inline src/Dynamo.All.sln:238]: you should also add this to the `mono.sln`

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    </Compile>
+    <Compile Include="SelectionTests.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\src\DynamoCore\DynamoCore.csproj">
+      <Project>{7858FA8C-475F-4B8E-B468-1F8200778CF8}</Project>
+      <Name>DynamoCore</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\..\..\src\Libraries\CoreNodeModels\CoreNodeModels.csproj">
```
Comment (mjkkirschner) [inline test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:61]: I think you are going to have the same trouble @RobertGlobant20 had - this should likely be marked as copy local false in visual studio.

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    <AssemblyName>CoreNodeModelsTests</AssemblyName>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <Deterministic>true</Deterministic>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>..\..\..\bin\AnyCPU\Debug\</OutputPath>
```
Comment (mjkkirschner) [inline test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:20]: these outputpaths should be set using a property:
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/DynamoCore.csproj#L49

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj (hunk 5 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:61]: ok I will fix that in the next commit, should I also change it for the other referenced packages or for CoreNodeModel only?

Comment (mjkkirschner) [reply test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:61]: checkout what binaries exist in the /nodes folder - you don't want to have any of those also copied to /bin after a build.

File: test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj (hunk 6 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/CoreNodeModelsTests.csproj:20]: great, also ready for the next commit

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        public void ClearSelectionTest()
+        {
+            //Selects the test node
+            selectionHelperMock
+                .Setup(x => x.RequestSelectionOfType("testMessage", SelectionType.Many, SelectionObjectType.Element))
+                .Returns(new List<ModelBase> { testNode });
+            selection.Select(testNode);
+
+            //checks one node is selected
+            var cant = selection.SelectionResults.Count();
```
Comment (aparajit-pratap) [inline test/Libraries/CoreNodeModelsTests/SelectionTests.cs:61]: Could you rename this to `count`.

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+                SerializeCore(nodeElement, context);
+
+            public void DeserializeCore(XmlElement nodeElement, SaveContext context) =>
+                DeserializeCore(nodeElement, context);
+
+            public bool UpdateValueCore(UpdateValueParams updateValueParams) =>
+                UpdateValueCore(updateValueParams);
+
+            public string GetOutputPortName() =>
+                 GetOutputPortName();
```
Comment (aparajit-pratap) [inline test/Libraries/CoreNodeModelsTests/SelectionTests.cs:109]: Why implement these when they're simply calling the base methods?

File: test/Libraries/CoreNodeModelsTests/app.config [hunk 9]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="DiffPlex" publicKeyToken="1d35e91d1bd7bc0f" culture="neutral" />
```
Comment (aparajit-pratap) [inline test/Libraries/CoreNodeModelsTests/app.config:6]: What's this for?

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs (hunk 7 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/SelectionTests.cs]: will do

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs (hunk 8 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/SelectionTests.cs]: the base methods are protected, this way I can call them for testing

File: test/Libraries/CoreNodeModelsTests/app.config (hunk 9 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/app.config]: I dont know, I will check whats up with that

Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/app.config]: That should be gone on the new commit

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs (hunk 7 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/SelectionTests.cs]: fixed on new commit

File: test/Libraries/CoreNodeModelsTests/SelectionTests.cs (hunk 8 above)
Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodeModelsTests/SelectionTests.cs]: also, you may notice some of this arent being tested yet, those tests are the ones im working on right now and will be added on a future commit soon

**Pull request merged**